### DISCUSSION TITLE
The error dialog box shown when trying to load a corrupted bitmap for the TeX Symbols is too unspecific

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#9] [1.2 Bugs and Issues]
Dear Alex, dear community,

When trying to load a corrupted bitmap file for the TeX Symbols, the following error dialog box is shown:
![grafik](https://github.com/WinEdt-Team/WinEdt-Support/assets/52357589/1c7ef544-505d-435a-a1b3-0787265b39f3)
I think that it would help if the dialog box would
- point out which ini file it could not load
- inform that the reason is a corrupted bitmap file
- give the name /  path to the file that is corrupted.

I had a hard time figuring out what was wrong since I had generated the bitmap file automatically and then copied it to the correct location and finally loaded GUI.ini. Apparently, as the process grew more complicated and took more time, I started copying the file while it was not yet completely generated, ending up with a corrupted file.
The zip file [MathZ1.zip](https://github.com/WinEdt-Team/WinEdt-Support/files/14129650/MathZ1.zip) should contain a corrupted bitmap file for testing (it is basically a text file whose extension I changed).

WinEdt Build: 20230519 (v. 11.1) - 64-bit
Operating system: Microsoft Windows 11 Home (10.0.22631 Build 22631)

Best,
Thomas

**Comment by Alex-Simonic** [comment]
Your suggestion might benefit error-prone users carelessly performing advanced customizations. It is not as simple as you might think.

The exception is raised by one of Delphi's bitmap handling methods (probably TBitmap.LoadFromFile(...)) as a result of a corrupted source. WinEdt code has to handle such exceptions by surrounding every potentially problematic call with:

try
  Call_Some_Function;
except
  Do something since an exception has been raised during the call
end;

WinEdt does have some handlers like this and that's why you got the message that an exception occurred while executing a macro LoadConfig (and WinEdt stayed operational). More such (nested) handlers would be needed to pinpoint the problem and provide a more precise message after an error like this occurs.

Alternatively, perhaps you should improve your automated process to check the bitmap integrity and synchronize the flow before loading corrupted or unfinished images into WinEdt. In other words, I will do nothing about this one as I don't consider it a WinEdt problem...

**Reply by Alex-Simonic** [reply]
This being said in the next version you will get a message:

Error Loading a GUI Group Bitmap from "[full path and filename]"

This should help you fix a problem...
